packages/bridge/src/runtime/composables.ts

```diff
@@ -250,7 +250,7 @@ export const abortNavigation = (err?: string | Partial<NuxtError>) => {
   throw err
 }
 
-type RouteMiddlewareReturn = void | Error | string | Location | boolean
+type RouteMiddlewareReturn = void | Error | string | Location | boolean | Route
```

Comment:
do we also need to accept undefined/null?

Reply:
Hmm, I investigated a bit and I don't think I have to accept it.